drivers/scd30/scd30.c

```diff
@@ -0,0 +1,379 @@
+ * @author      Jan Schlichter   <schlichter@ibr.cs.tu-bs.de>
+ * @}
+ */
+
+#include <string.h>
+#include "periph/i2c.h"
+#include "xtimer.h"
+#include "byteorder.h"
+#include "checksum/crc8.h"
+
+#include "scd30.h"
+#include "scd30_internal.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define SCD30_I2C           (dev->params.i2c_dev)
+#define SCD30_I2C_ADDRESS   (dev->params.i2c_addr)
+
+static int8_t _scd30_read_data(scd30_t *dev, scd30_measurement_t *result);
+static int _scd30_crc_check(uint8_t *buff, uint8_t len);
+static bool _is_valid_interval(uint16_t *interval);
+static bool _is_valid_press_comp(uint16_t *apc);
+static inline float _reinterpret_float(uint32_t raw_val);
+static float _raw_val_to_float(const uint8_t *buffer);
+
+int8_t scd30_init(scd30_t *dev, const scd30_params_t *params)
+{
+    DEBUG("[scd30] init\n");
+    dev->params = *params;
+
+    int ret = 0;
+
+    uint16_t version;
+    ret = scd30_get_param(dev, SCD30_VERSION, &version);
+    DEBUG("[scd30] --- Version 0x%02x%02x\n", (version >> 8), version);
+    if (ret < 0) {
+        DEBUG("[scd30] Failed to get version\n");
+        return ret;
+    }
+
+    DEBUG("[scd30] init: done.\n");
+
+    return SCD30_OK;
+}
+
+int8_t scd30_set_param(scd30_t *dev, uint16_t param, uint16_t val)
+{
+    uint8_t buffer[5];
+    int ret = 0;
+
+    buffer[0] = (uint8_t)(param >> 8);
+    buffer[1] = (uint8_t)param;
+    buffer[2] = (uint8_t)(val >> 8);
+    buffer[3] = (uint8_t)val;
+    buffer[4] = crc8(buffer + 2, 2, SCD30_CRC_FUNC, SCD30_CRC_START_VAL);
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 5, 0);
+    i2c_release(SCD30_I2C);
+    if (ret != 0) {
+        return SCD30_COM_FAILED;
+    }
+    else {
+        xtimer_usleep(SCD30_READ_WRITE_SLEEP_US);
+        return SCD30_OK;
+    }
+}
+
+int8_t scd30_get_param(scd30_t *dev, uint16_t param, uint16_t *val)
+{
+    uint8_t buffer[3];
+    int ret = 0;
+
+    buffer[0] = (uint8_t)(param >> 8);
+    buffer[1] = (uint8_t)param;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    if (ret != 0) {
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    i2c_release(SCD30_I2C);
+
+    xtimer_usleep(SCD30_READ_WRITE_SLEEP_US);
+
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_read_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 3, 0);
+    i2c_release(SCD30_I2C);
+
+    if (ret != 0) {
+        DEBUG("[scd30] Problem reading param %u from sensor\n", param);
+    }
+
+    if (_scd30_crc_check(buffer, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+
+    *val = (buffer[0] << 8) | buffer[1];
+    return SCD30_OK;
+}
+
+int8_t scd30_read_triggered(scd30_t *dev, scd30_measurement_t *result,
+                            uint16_t *apc)
+{
+    uint16_t state = 0;
+    uint16_t curr_interval;
+    uint32_t initial_time;
+    int ret;
+
+    if (!_is_valid_press_comp(apc)) {
```

Comment:
This is the only use of `apc` in this function.

Reply:
Yeah, I just realised that this is from an earlier implementation and I just didn't remove this part. Will remove it in the fixup.